### PR TITLE
make shufflenet and resnet scriptable

### PR DESCRIPTION
**Pull request by eellison**
ShuffleNet:
- added type annotations 
- previously `branch1` was only conditionally defined

Resnet:
- need to add `downsample` to constants so that script can compile when it is None

### CODE REVIEW

File: torchvision/models/resnet.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def conv1x1(in_planes, out_planes, stride=1):
 
 class BasicBlock(nn.Module):
     expansion = 1
+    __constants__ = ['downsample']
```
Comment (fmassa) [inline torchvision/models/resnet.py]: Can you explain a bit why this is a `constant`? `downsample` has learnable parameters.

Comment (fmassa) [reply torchvision/models/resnet.py]: Ok, I think I got it. I think I prefer the `__constants__` in this case, even though `Optional[Module]` would be nicer.

**Comment by fmassa** [comment]
CI is failing, probably because we might need to use PyTorch nightly now?

Can you try changing https://github.com/pytorch/vision/blob/0bd7080c6fa4d8a07f34aa66caf10683824e5b82/.travis.yml#L29 and see if tests work now?

**Comment by fmassa** [comment]
I've merged another PR that changed the CI to use pytorch nightly, so let me re-run the CI

**Comment by eellison** [comment]
@fmassa thanks!

**Comment by fmassa** [comment]
@eellison there is a linter error due to the type annotation not finding `Tensor`. Can you look into fixing it, either by doing `torch.Tensor` or by making the linter less strict?

**Comment by eellison** [comment]
> @eellison there is a linter error due to the type annotation not finding `Tensor`. Can you look into fixing it, either by doing `torch.Tensor` or by making the linter less strict (or something else)?

yep will do

File: torchvision/models/shufflenetv2.py [hunk 2]
```diff
@@ -51,6 +52,8 @@ def __init__(self, inp, oup, stride):
                 nn.BatchNorm2d(branch_features),
                 nn.ReLU(inplace=True),
             )
+        else:
+            self.branch1 = nn.Sequential()
```
Comment (fmassa) [inline torchvision/models/shufflenetv2.py]: I'm merging this in, but it would be great if torchscript could support `Optional[nn.Module]`.
